ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/recovery/ExtendedRecoveryModule.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.arjuna.ats.arjuna.recovery;
+
+import java.util.Set;
+
+public interface ExtendedRecoveryModule extends RecoveryModule {
```

Comment:
okay

Reply:
Although, one could argue that the only resources supported on EAP are ones that get wrapped by JCA so this interface may still make sense.

Also notice that there is javadoc for each individual method in the interface.

Reply:
I was probably not clear, I'm sorry. My point was about adding a javadoc with information "why this was created". As you mentioned there are docs at the methods but they describes what they do, just I think it would by good to note "why".

Reply:
Periodic recovery is for developers and they should not need hand holding. One does not produce use cases in javadoc, those are for design docs or for JIRA explanations.
If the dev does want to see a use case then he just needs to  perform a find usages to see where it is used.